### DISCUSSION TITLE
Error in rbind

### DISCUSSION THREAD

**Discussion by mikatakahashi** [IBAHCM/RPiR#182] [Archive] [answered]
Hi @soniamitchell @richardreeve 
I get error message saying "Error in rbind(deparse.level, ...) : numbers of columns of arguments do not match" after running 

latest.population <- population.df
while (latest.population$time < end.time) {
  latest.population <- timestep_deterministic_SIS(latest = latest.population,
                                                  transmission.rate = ecoli.transmission, 
                                                  recovery.rate = ecoli.recovery, 
                                                  timestep = this.timestep)
  population.df <- rbind(population.df, latest.population) 
}

on practical 2-4. Could you please help me solve it?

**Comment by soniamitchell** [comment, marked as answer]
Hey @2591129, it looks like (on the last line you pasted) population.df and latest.population have different numbers of columns and rbind() doesn’t like that. Try looking at the objects to confirm this for yourself. Then try to work out where these objects were created and what happens to them in your code. If you still have problems it might be easier to help in your repo, where we can point to particular lines. Let us know how you get on :)

**Reply by mikatakahashi** [reply]
So is this a problem with my function I created in step file? I assigned "latest.population <- population.df" at very first so they should have same column numbers.

**Reply by soniamitchell** [reply]
I've just run your code and I'd say yes. It might help you see the problem a little better if you try this.

As you observed,

```
population.df <- data.frame(time = start.time,
                      susceptibles = initial.susceptibles,
                      infecteds = initial.infecteds)

latest.population <- population.df
while (latest.population$time < end.time) {
  latest.population <- timestep_deterministic_SIS(latest = latest.population,
                                                  transmission.rate = ecoli.transmission, 
                                                  recovery.rate = ecoli.recovery, 
                                                  timestep = this.timestep)
  population.df <- rbind(population.df, latest.population) 
}
```

results in an error:

```
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```

If you look at `latest.population`, it looks like this:

```
     susceptibles infecteds
1     96.70667  3.293333
```

whereas `population.df` looks like this:

```
     time susceptibles infecteds
1    0           98         2
```

The clues in the exercise are:
> And the `timestep_deterministic_SIS()` function can update the time along with the number of susceptibles and infecteds. Here, it makes the code slightly more elegant but it might be critical if, for instance, we wanted to adjust the timestep as we went along, which we will need to do later.

and 

> latest.population$time is the time that we are currently at, which we extract from the data frame after each timestep.

Does that help?

**Reply by mikatakahashi** [reply]
Ah, so on the very last line within function, it needs to return time to the data.frame as well as susceptible and infected populations? That makes sense.

**Reply by soniamitchell** [reply]
Yep. After editing your code and running through your loop. Remember to look at the objects and make sure they look like you expect them to 👍

**Reply by mikatakahashi** [reply]
Thank you!
